source/agora/node/Ledger.d

```diff
@@ -229,6 +235,57 @@ public class Ledger
         else
             return null;
     }
+
+    /***************************************************************************
+
+        Write a block to the ledger file on disk.
+
+        Params:
+            fd = Instance of LedgerFile
+            block = Type of struct to writeFile
+
+    ***************************************************************************/
+
+    public void writeBlockFile (ref File fd, ref Block block) @trusted
```

Comment:
You do not have to pass the file by reference. The `File` struct has an internal reference counter, which means it acts as a reference. So you can pass it by value.

Also could you rename it to `File file;`? It makes it a bit clearer, it's not just a file descriptor but a wrapper around it.

Reply:
Fixed